velox/exec/WindowPartition.cpp

```diff
@@ -383,6 +384,20 @@ void WindowPartition::updateKRangeFrameBounds(
             orderByRowColumn.nullByte(),
             orderByRowColumn.nullMask()));
 
+    // Mark the frame invalid if the frame bound is NaN.
+    auto frameType = data_->columnTypes()[mappedFrameColumn];
+    if (frameType->isReal()) {
```

Comment:
We could avoid this check on a per row basis. They can be computed outside the loop or the outer function could pass to the inner function if it should do a float or double NaN check.